### ISSUE TITLE
equation graph needs update to handle new global equations

### ISSUE THREAD

**Issue by jzacsh** [rxie25/PLT2016Spring#94]
global equations (#6) was designed and implemented in scanner/parser after equation relationships (from #56)  were finished. Need to go back into eq's code and make sure there's a pre-pass for global contexts.

the work here is similar to pull request #93. currently we do analysis in this order:
 1. analyze equations in each context decl
 2. analyze equations in each `find` decl

**tl;dr** need to modify the loop that walks over contexts in step 1 to:
 - **include** global contexts
 - have have **normal** contexts inherit the global relationships

**Comment by jzacsh** [comment]
So, interestingly, the way #6 was solved, tianci pointed out that each global context is "copy/pasted"~ish (injected) into each regular context's AST.

**tl;dr** automagically, the following program _(side-by-side with its relevant sast)_ correctly prints `42`:
```
$ pr -m tests/test-global-variable-deps.eq <(./eqeq.native -s < tests/test-global-variable-deps.eq | sed '/^$/d') 

a = 40;                             Global : {
                                        deps:
SomeCtx = {                             indeps:
  b = 2 + a;                                a : ```
}                                               40.;
                                            ```
SomeCtx: find b {                       finds:
  print("%.0f\n", b);               }
}                                   SomeCtx : {
                                        deps:
                                            b: [a]
                                        indeps:
                                            a : ```
                                                40.;
                                            ```
                                        finds:
                                            "find_SomeCtx_0": {
                                                [0]: {
                                                    deps:
                                                        b: [a]
                                                    indeps:
                                                        a : ```
                                                            40.;
                                                        ```
                                                }
                                            }
                                    }

```

Leaving this assigned to @tiancizhong who said she'd add some more tests to ensure this really works out.
